lib/sidekiq/reliable_fetcher.rb

```diff
@@ -12,20 +13,35 @@ def self.setup_reliable_fetch!(config)
     end
 
     def initialize(options)
-      @strictly_ordered_queues = !!options[:strict]
-      @queues = options[:queues].map { |q| "queue:#{q}" }
-      @unique_queues = @queues.uniq
-      @last_clean = Time.now.to_i
+      queues = options[:queues].map { |q| "queue:#{q}" }
+
+      @unique_queues = queues.uniq
+      @queues_iterator = queues.shuffle.cycle
+      @queues_size  = queues.size
+
+      @nb_fetched_jobs = 0
       @cleaning_interval = options[:cleaning_interval] || DEFAULT_CLEANING_INTERVAL
       @consider_dead_after = options[:consider_dead_after] || DEFAULT_DEAD_AFTER
     end
 
     def retrieve_work
-      clean_working_queues! if @cleaning_interval != -1 && Time.now.to_i - @last_clean > @cleaning_interval
+      clean_working_queues! if @cleaning_interval != -1 && @nb_fetched_jobs >= @cleaning_interval
+
+      for i in 0..@queues_size
```

Comment:
Ici tu as itérer n+1 fois pour une file de n éléments ; je pense que tu veux itérer n fois.

Comme ça plutôt ?
```ruby
q.size.times do
  puts i
end
```